wkcuber/image_readers.py

```diff
@@ -150,34 +154,54 @@ def find_count_of_axis(tif_file: TiffFile, axis: str) -> int:
 
     def read_array(self, file_name: str, dtype: np.dtype, z_slice: int) -> np.ndarray:
         with TiffFile(file_name) as tif_file:
-            num_channels = self.read_channel_count(file_name)
-            if len(tif_file.pages) > num_channels:
+            if self.num_channels is None:
+                self.num_channels = self.read_channel_count(file_name)
+            if self.is_page_multi_channel is None:
+                # we assume all tif pages have the same dimensions
+                self.is_page_multi_channel = tif_file.pages[0].ndim == 3
+
+            if self.is_page_multi_channel:
+                self.num_channels = 1
```

Comment:
Why are `num_channels` and `is_page_multi_channel` stored as an instance variable?

Reply:
I wanted to "cache" these values for future calls.

Reply:
I am not as deep in the `ImageReader` as you are, but my understanding is that there is a single instance for managing the image readers (`image_reader = ImageReaderManager()`).  The manager has a dict with an instance for each different type (e.g. `TiffImageReader`). If I now want to read two completely different images (both .tiff), wouldn't this be a problem because the `num_channels` could differ?

Reply:
Yes, but for the cubing case we assume that all files have similar dimensions.
If we want to use the cubing for two different datasets, the cuber is executed twice, so that shouldn't be a problem. 

Are there places, where only parts of the cuber code are executed?

Reply:
I only found the `image_reader` in connection with `cubing`. Then it should be fine.